Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/TextInput.swift

```diff
@@ -134,7 +136,23 @@ private extension TextInput {
     /// The keyboard type to use depending on where the input is numeric and decimal.
     var keyboardType: UIKeyboardType {
         guard let fieldType = element.fieldType else { return .default }
-        return fieldType.isNumeric ? (fieldType.isFloatingPoint ? .decimalPad : .numberPad) : .default
+        
+        if fieldType.isNumeric {
+#if os(visionOS)
+            // The 'positiveNegativeButton' doesn't show on visionOS
+            // so we need to show this keyboard so the user can type
+            // a negative number.
+            return .numbersAndPunctuation
+#else
+            if fieldType.isFloatingPoint {
+                return .decimalPad
+            } else {
+                return .numberPad
+            }
+#endif
+        } else {
+            return .default
+        }
```

Comment:
You could simplify this using an `if` expression.

```suggestion
        return if fieldType.isNumeric {
#if os(visionOS)
            // The 'positiveNegativeButton' doesn't show on visionOS
            // so we need to show this keyboard so the user can type
            // a negative number.
            .numbersAndPunctuation
#else
            if fieldType.isFloatingPoint { .decimalPad } else { .numberPad }
#endif
        } else {
            .default
        }
```